docs/defining-models.md

```diff
@@ -20,49 +19,81 @@ The `defmodel` macro is used to define a new model:
+
+``` clojure
+User     ;;=> #my.ns.UserInstance{:table :user_table, :name "User", :toucan.models/model true}
+(User 1) ;;=> #my.ns.UserInstance{:id 1, :first_name "Rapid", :last_name "Raccoon"}
+```
+
+This way they can share the `IModel` protocol (discussed below) and simplifies the interface; functions like `types`
+work on either the model itself (e.g., `User`) or instances fetched from the DB.
 
 In other words, both of the following work:
 
-```clojure
+``` clojure
 ;; call on the model definition
 (models/types User) ; -> {:status :keyword}
 
 ;; call on an instance of the model
 (models/types (User 1)) ; -> {:status :keyword}
 ```
 
+## Implementing protocols and interfaces
+
+The `defmodel` macro is similar to `defrecord` or `deftype`, in that you can make the new type implement specific
+protocols or interfaces.
+
+``` clojure
+(defmodel User :user_table
+  MyProtocol
+  (do-a-thing [this]
+    ,,,))
+```
 
 ## The IModel Interface
 
-The `IModel` protocol defines the various methods that are used to provide custom behavior for various models.
-All models defined by `defmodel` automatically implement this protocol using the default definitions in `IModelDefaults`,
-but you can override one or more of the methods to suit your needs.
+The `IModel` protocol defines the various methods that are used to provide custom behavior for various models. All
+models defined by `defmodel` automatically implement this protocol using the default definitions in `IModelDefaults`,
+but you can override one or more of the methods to suit your needs. Any method you don't specify will get the default
+implementation found in `IModelDefaults`.
+
+``` clojure
+(defmodel User :user_table
+  IModel
+  (hydration-keys [_]
+    [:user])
+  (properties [_]
+    {:timestamped true})
+  (pre-insert [user]
+    user))
+```
 
-To override one or more implementations, use `extend` and `merge` your custom implementations with `IModelDefaults`:
+This is equivalent to using `extend` manually, merging your custom implementations with `IModelDefaults`:
 
 ```clojure
+(defmodel User :user_table)
+
 (extend (class User)             ; it's somewhat more readable to write `(class User)` instead of `UserInstance`
   IModel (merge IModelDefaults
                 {:types (constantly {:status :keyword}))
 ```
 
-Note that `User` itself is not a class, but rather a definition of the `User` model, and is itself an instance of `UserInstance`;
-hence the call to `extend` `(class User)`.
+Note that `User` itself is not a class, but rather a definition of the `User` model, and is itself an instance of
+`UserInstance`; hence the call to `extend` `(class User)`.
 
-Refer to the documentation below for an overview of the various methods provided by `IModel` and how to use them to customize
-the behavior of your models.
+Refer to the documentation below for an overview of the various methods provided by `IModel` and how to use them to
+customize the behavior of your models.
```

Comment:
again, random edits like this that don't affect the way the markdown renders make this PR a lot bigger (and thus a lot harder to review), so what's the reason behind them?